### ISSUE TITLE
Setting detach to false on the docker_container resource guarantees Docker::Error::TimeoutError will be raised

### ISSUE THREAD

**Issue by keiths-osc** [chef-cookbooks/docker#694]
Starting with version 2.6.1 through the current version (2.6.6 at the time this was logged), setting the `detach` property to `false` on the `docker_container` resource and executing the `run` action seems guaranteed to cause an error where it previously did not.

Here is a small recipe that demonstrates the problem.  (This was executed on centos 7.1 using chef 12.8.1).  When I run this:

```
include_recipe 'yum-docker'
include_recipe 'yum'

docker_service 'default' do
  install_method 'package'
  action [:create, :start]
end

docker_image 'ubuntu' do
  tag 'latest'
end

docker_container 'test' do
  image 'ubuntu'
  tag 'latest'
  command 'ls'
  detach false
end
```

The result is this:

```
================================================================================
Error executing action `run` on resource 'docker_container[test]'
================================================================================

Docker::Error::TimeoutError
---------------------------
Container test failed to change to running state after 20 seconds

Cookbook Trace:
---------------
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/docker/libraries/helpers_container.rb:88:in `wait_running_state'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/docker/libraries/docker_container.rb:293:in `block (2 levels) in <class:DockerContainer>'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/docker/libraries/docker_container.rb:288:in `block in <class:DockerContainer>'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:122:in `instance_eval'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:122:in `compile_and_converge_action'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/docker/libraries/docker_container.rb:159:in `call_action'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/docker/libraries/docker_container.rb:322:in `block in <class:DockerContainer>'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:122:in `instance_eval'
/tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/compat_resource/files/lib/chef_compat/copied_from_chef/chef/provider.rb:122:in `compile_and_converge_action'

Resource Declaration:
---------------------
# In /tmp/vagrant-chef/70a9f794f1b949666db51c8d70dc8d4d/cookbooks/dockertest/recipes/default.rb

 13: docker_container 'test' do
 14:   image 'ubuntu'
 15:   tag 'latest'
 16:   command 'ls'
 17:   detach false
 18: end
```

The container itself ran fine and exited normally:

```
[root@centos71 ~]# docker ps -a
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS                      PORTS               NAMES
7f61d89cd621        ubuntu:latest       "ls"                12 minutes ago      Exited (0) 12 minutes ago                       test
```

When I run the same recipe using version 2.6.0, the error does not occur.

When `detach` is `false`, then [this line](https://github.com/chef-cookbooks/docker/blob/v2.6.6/libraries/docker_container.rb#L291) waits for the container to exit.  Once the container exits, the [wait_running_state method](https://github.com/chef-cookbooks/docker/blob/v2.6.6/libraries/helpers_container.rb#L78-L89) is called and complains because the container is not running.  However, it's not _supposed_ to be in this case.

**Comment by chasebolt** [comment]
fixed in v2.6.7

**Comment by pnsaraf** [comment]
I am facing the same issue with cookbook 2.6.8. Even if i give detach as true its the same.
My default.rb

docker_service 'default' do
  action [:create, :start]
end

docker_image 'ubuntu' do
  action :pull
end

docker_container 'ubuntu1r' do
  repo 'busy box'
 detach true
  command "nc -ll -p 1234 -e /bin/cat"
end
